File: examples/real-world/string_utils.c
// String utility functions inspired by real C codebases
// Testing common patterns: string length, comparison, copying

int string_length(char* str) {
    int len;
    len = 0;
    while (*str != 0) {
        len = len + 1;
        str = str + 1;
    }
    return len;
}

int string_compare(char* s1, char* s2) {
    while (*s1 != 0 && *s2 != 0) {
        if (*s1 != *s2) {
            return *s1 - *s2;
        }
        s1 = s1 + 1;
        s2 = s2 + 1;
    }
    return *s1 - *s2;
}

void string_copy(char* dest, char* src) {
    while (*src != 0) {
        *dest = *src;
        dest = dest + 1;
        src = src + 1;
    }
    *dest = 0;
}

File: examples/real-world/string_utils.rs
static mut ERRNO: i32 = 0;
fn string_length(str: &[u8]) -> i32 {
    let mut str_idx: usize = 0;
    let mut len: i32 = 0i32;
    len = 0;
    while str[str_idx] != 0 {
    len = len + 1;
    str_idx += 1 as usize;
}
    return len;
}
fn string_copy(dest: &mut [u8], src: &[u8]) {
    let mut dest_idx: usize = 0;
    let mut src_idx: usize = 0;
    while src[src_idx] != 0 {
    dest[dest_idx] = src[src_idx];
    dest_idx += 1 as usize;
    src_idx += 1 as usize;
}
    dest[dest_idx] = 0;
}
fn string_compare(s1: &[u8], s2: &[u8]) -> i32 {
    let mut s1_idx: usize = 0;
    let mut s2_idx: usize = 0;
    while (s1[s1_idx] != 0) && (s2[s2_idx] != 0) {
    if s1[s1_idx] != s2[s2_idx] {
    return (s1[s1_idx] as i32) - (s2[s2_idx] as i32);
}
    s1_idx += 1 as usize;
    s2_idx += 1 as usize;
}
    return (s1[s1_idx] as i32) - (s2[s2_idx] as i32);
}
